diff --git a/compiler/rustc_target/src/spec/i586_unknown_linux_gnu.rs b/compiler/rustc_target/src/spec/i586_unknown_linux_gnu.rs
index 7c92dda8a9d..09abaf91c28 100644
--- a/compiler/rustc_target/src/spec/targets/i586_unknown_linux_gnu.rs
+++ b/compiler/rustc_target/src/spec/targets/i586_unknown_linux_gnu.rs
@@ -2,7 +2,7 @@
 
 pub fn target() -> Target {
     let mut base = super::i686_unknown_linux_gnu::target();
     base.rustc_abi = None; // overwrite the SSE2 ABI set by the base target
-    base.cpu = "pentium".into();
+    base.cpu = "pentium3".into();
     base.llvm_target = "i586-unknown-linux-gnu".into();
     base
 }
